odoo/addons/base/models/ir_actions_report.py

```diff
@@ -554,6 +601,14 @@ def barcode(self, barcode_type, value, **kwargs):
             if kwargs['quiet']:
                 kwargs['barBorder'] = 0
 
+        if barcode_type in ('EAN8', 'EAN13') and not self.check_barcode_encoding(value, barcode_type.lower()):
```

Comment:
I think in `master`, it could be cool to remove the `lower` and instead write the `barcode_sizes` key in uppercase as we will have to replace the calls to `barcode.nomenclature` `check_encoding` by `ir.actions.report` `check_barcode_encoding` (so we can rewrite the passed params too)

Reply:
Fully agreed